build.rs

```diff
@@ -134,10 +185,180 @@ fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, St
     config_map
 }
 
-fn load_cross_compile_info() -> Result<(InterpreterConfig, HashMap<String, String>)> {
-    let python_include_dir = env::var("PYO3_CROSS_INCLUDE_DIR")?;
-    let python_include_dir = Path::new(&python_include_dir);
+/// Parse sysconfigdata file
+///
+/// The sysconfigdata is basically a dictionary, and since we can't really use this library to read
+/// it for us, egg and chicken type thing, we parse it with some regex. Here there are two regex's.
+/// The first one is to match an entry in the dictionary (`entry_re`). Then when the entry is on
+/// multiple lines we use the second regex to capture the additional string entry and add it to the
+/// previously captured key. We detect if this is a multi line entry by checking if the last capture
+/// group contains either `,` or `}`.
+///
+/// ## entry_re
+///
+/// The first part `'([a-zA-Z_0-9]*)'` we match a key in the dictionary and extract it without the quotes,
+/// The second capture group is `((?:"|')([\S ]*)(?:"|')|\d+)` we can split this capture group into 2,
+///
+/// * `(?:"|')([\S ]*)(?:"|')` here we have non capturing groups for the quotes `(?:'|")` and the
+///   capture group which matches all non-whitespace characters and any normal space characters.
+///   Since this is a capture group within another capture group, this is considered the third group
+///   and might be None if the value is not a string.
+/// * `\d+` matches any one or more digit characters.
+///
+/// The last capture `($|,|})`group allows us to check if the entry is finished or not. This is done
+/// by matching either:
+/// * The end of line `$`
+/// * The `,` to allow the entry of a new key
+/// * The end of the dictionary `}`
+///
+/// ## subsequent_re
+///
+/// Here we have the same string matching group as we have in `entry_re` with the end of entry detection.
+/// The only addition to this sub-regex is the matching of one or more whitespace characters `\s+`.
+///
+/// # Example matches
+///
+/// ```py
+/// build_time_vars = {'ABIFLAGS': 'm',
+/// 'AC_APPLE_UNIVERSAL_BUILD': 0,
+/// 'AIX_GENUINE_CPLUSPLUS': 0,
+/// 'ANDROID_API_LEVEL': 26,
+/// 'AR': 'arm-linux-androideabi-ar',
+/// 'ARFLAGS': 'rcs',
+/// 'BASECFLAGS': '-mfloat-abi=softfp -mfpu=vfpv3-d16 -Wno-unused-result '
+///               '-Wsign-compare -Wunreachable-code',
+/// 'BASECPPFLAGS': '-IObjects -IInclude -IPython'}
+/// ```
+#[cfg(feature = "cross-compile")]
+fn parse_sysconfigdata(config_path: impl AsRef<Path>) -> Result<HashMap<String, String>> {
+    let config_reader = BufReader::new(File::open(config_path)?);
```

Comment:
How about making Python output only what we need?
E.g,
```rust
    let mut script = fs::read_to_string(config_path)?;
    s += r#"
print("version_major", build_time_vars["VERSION"][0])  # 3
print("version_minor", build_time_vars["VERSION"][2])  # E.g., 8
...
"#;
    let config = run_python_script(find_interpreter()?, script)?;
```
(Note that  I introduced `find_interpreter` in #1097)

Then we can use the same code as `get_config_from_interpreter`.

Reply:
It's possible when cross-compiling we may not have a Python interpreter for the host machine which can run the script? Though this approach is also quite nice.

Reply:
This would definitively be cleaner, with less complicated regex's. I would be curious to see which environments dont have a host python interpreter when cross compiling python. For example, the build system to compile python for android (buildozer) first compiles the same version of python for the host before cross-compiling python for the target.

Reply:
applied